File: frontend/lib/models/date_plan.dart
class DatePlan {
  final int id;
  final int relationshipId;
  final int creatorId;
  final String title;
  final DateTime date;
  final String? location;
  final String status; // 'pending', 'accepted', 'declined'

  DatePlan({
    required this.id,
    required this.relationshipId,
    required this.creatorId,
    required this.title,
    required this.date,
    this.location,
    required this.status,
  });

  factory DatePlan.fromJson(Map<String, dynamic> json) {
    return DatePlan(
      id: json['id'],
      relationshipId: json['relationship_id'],
      creatorId: json['creator_id'],
      title: json['title'],
      date: json['date'] != null 
          ? DateTime.parse(json['date']) 
          : DateTime.now(),
      location: json['location'],
      status: json['status'] ?? 'pending',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'relationship_id': relationshipId,
      'creator_id': creatorId,
      'title': title,
      'date': date.toIso8601String(),
      'location': location,
      'status': status,
    };
  }
}

File: backend/app/Models/DatePlan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Attributes\Fillable;

#[Fillable(['relationship_id', 'creator_id', 'title', 'date', 'location', 'status'])]
class DatePlan extends Model
{
    protected function casts(): array
    {
        return [
            'date' => 'datetime',
        ];
    }

    public function relationship(): BelongsTo
    {
        return $this->belongsTo(Relationship::class);
    }

    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'creator_id');
    }
}
